src/main/java/org/apache/kafka/server/quota/StaticQuotaCallback.java

```diff
@@ -61,6 +79,52 @@ public void close() {
     public void configure(Map<String, ?> configs) {
         StaticQuotaConfig config = new StaticQuotaConfig(configs, true);
         quotaMap = config.getQuotaMap();
-        log.info("Configured quota callback with {}", quotaMap);
+        storageQuota = config.getStorageQuota();
+        storageCheckInterval = config.getStorageCheckInterval();
+        logDirs = config.getLogDirs();
+
+        try {
+            storageCheckerThread.interrupt();
+            storageCheckerThread.join();
+            storageCheckerThread.start();
```

Comment:
This doesn't look right. Normally `configure()` is only called once (there's a separate `Reconfigurable` for things which are explicitly reconfigurable). So there shouldn't be any need for `interrupt()` and `join()` AFAICS. Did I miss something?

Also `close()` should cause the thread to finish, likely blocking until it has finished.

Reply:
Good point on the close. Regarding (re)configure, you're right, but I just wrote the logic with having it deal with the reconfigure anyway, because I wasn't sure if it is possible to make it reconfigurable or not.

Reply:
Quoting from the javadoc for `Thread.start()`:

> It is never legal to start a thread more than once.
 
and

> @throws     IllegalThreadStateException  if the thread was already started.

So I think you should just call start here, arrange for the thread to stop in `close()` and rely on the life cycle implied (but not exactly spelled out explictly) by the plugin contract.